src/software/thunderscope/chicker/chicker.py

```diff
@@ -0,0 +1,186 @@
+    QPushButton,
+    QRadioButton,
+    QVBoxLayout,
+    QHBoxLayout,
+    QWidget,
+    QSlider,
+)
+
+
+class ChickerWidget(QWidget):
+    def __init__(self, parent=None):
+        super(ChickerWidget, self).__init__(parent)
+
+        """
+        The Chicker Widget grid is laid out in the following way:
+                    ┌────────┐     ┌──────┐      ┌──────┐
+                    │ Charge │     │ Kick │      │ Chip │
+                    └────────┘     └──────┘      └──────┘
+
+                    ┌────────┐   ┌──────────┐ ┌─────────┐
+                    │No Auto │   │Auto Kick │ │Auto Chip│
+                    └────────┘   └──────────┘ └─────────┘
+
+                    ┌───────────────────────────────────┐
+                    │           Geneva Slider           │
+                    └───────────────────────────────────┘
+
+                    ┌───────────────────────────────────┐
+                    │           Power Slider            │
+                    └───────────────────────────────────┘
+        """
+        # grid --> groupBox --> button/slider
+        grid = QGridLayout()
+        self.radio_buttons = QButtonGroup()
+
+        self.charge_button_list = common_widgets.create_button("Charge", False)
+        self.charge_button = self.charge_button_list[1]
+        grid.addWidget(self.charge_button_list[0], 0, 0)
+
+        self.kick_button_list = common_widgets.create_button("Kick")
+        self.kick_button = self.kick_button_list[1]
+        grid.addWidget(self.kick_button_list[0], 0, 1)
+
+        self.chip_button_list = common_widgets.create_button("Chip")
+        self.chip_button = self.chip_button_list[1]
+        grid.addWidget(self.chip_button_list[0], 0, 2)
+
+        self.no_auto_list = common_widgets.create_radio("No Auto", self.radio_buttons)
+        self.no_auto_button = self.no_auto_list[1]
+        grid.addWidget(self.no_auto_list[0], 1, 0)
+
+        self.auto_kick_list = common_widgets.create_radio(
+            "Auto Kick", self.radio_buttons
+        )
+        self.auto_kick_button = self.auto_kick_list[1]
+        grid.addWidget(self.auto_kick_list[0], 1, 1)
+
+        self.auto_chip_list = common_widgets.create_radio(
+            "Auto Chip", self.radio_buttons
+        )
+        self.auto_chip_button = self.auto_chip_list[1]
+        grid.addWidget(self.auto_chip_list[0], 1, 2)
+
+        self.geneva_slider_list = common_widgets.create_slider("Geneva Slider", 1, 5, 1)
+        self.geneva_slider = self.geneva_slider_list[1]
+        grid.addWidget(self.geneva_slider_list[0], 2, 0, 1, 3)
+
+        self.power_slider_list = common_widgets.create_slider(
+            "Power Slider", 1, 100, 10
+        )
+        self.power_slider = self.power_slider_list[1]
+        grid.addWidget(self.power_slider_list[0], 3, 0, 1, 3)
+
+        self.setLayout(grid)
+        self.grid = grid
+        # state of radio buttons in order NoAuto, AutoKick, AutoChip
+        self.radio_checkable = [True, True, True]
```

Comment:
Can you make this a map so its more clear where its used

so it looks like `self.radio_checkable["no_auto"] = False`